### ISSUE TITLE
HMR - Component Duplicated on Reload

### ISSUE THREAD

**Issue by pfazzino** [DeMoorJasper/parcel-plugin-svelte#137]
Just started testing Parcel using Svelte. Wanted to use HMR so added

`if (module.hot) {
  module.hot.accept()
}`

to main.js (doesn't matter where I add this) but got the same symptoms as that reported in #31 with output being appended on save. It's not just when editing the main.js though, any save on the svelte component causes the output to be appended.

See example - https://github.com/pfazzino/parcelsveltetest

**Comment by benhormann** [comment]
```
if (module.hot) {
  module.hot.dispose(() => {
    app.$destroy();
  });
  module.hot.accept();
}
```
Credit: figured this out from rixo's templates <https://github.com/rixo/svelte-hmr>.

**Comment by pfazzino** [comment]
Yup good fix. This isn't full hmr as state isn't preserved between reloads but it's still a valuable feature. Is the lack of state preservation a limitation of svelte itself until hmr is officially supported https://github.com/sveltejs/svelte/issues/3632

**Comment by benhormann** [comment]
Kinda, rixo has got hmr working well enough in webpack / rollup, and even experimenting with snowpack.

Stores are preserved even though component state isn't.
